lib/Cro/Policy/Timeout.pm6

```diff
@@ -0,0 +1,59 @@
+class X::Cro::Policy::Timeout::InvalidTimeout is Exception {
+    has $.kind;
+    has @.kinds;
+
+    method message {
+        "Invalid kind of timeout, expected one of [@!kinds.join(', ')], got: $!kind, ignoring";
+    }
+}
+
+class X::Cro::Policy::Timeout::InvalidTimeoutValue is Exception {
+    has $.kind;
+    has $.value;
+
+    method message {
+        "Invalid timeout value passed for $!kind, got: $!value, ignoring";
+    }
+}
+
+role X::Cro::Policy::Timeout is Exception {
+    has $.phase;
+}
+
+role Cro::Policy::Timeout[%phase-defaults] {
+    has Real() $.total is required;
+    has Real() %.phases;
+
+    submethod BUILD(Real:D() :$!total, *%phases --> Nil) {
+        for %phase-defaults.kv -> $kind, $default {
+            my $supplied-value = %phases{$kind}:delete;
+            if $supplied-value ~~ Real {
+                %!phases{$kind} = Real($supplied-value);
+            } else {
+                %!phases{$kind} = Real($default);
+                with $supplied-value {
+                    warn X::Cro::Policy::Timeout::InvalidTimeoutValue.new(value => $_, :$kind);
```

Comment:
Given this indicates an implementation error, I wonder if we'd be better to `die` rather than `warn`?

Reply:
We have `warn` for excessive keys, so I thought it fits here too. Changed.

Reply:
Ah, I had thought to maybe `die` for both.